python/_tskitmodule.c

```diff
@@ -9860,6 +9860,128 @@ TreeSequence_divergence_matrix(TreeSequence *self, PyObject *args, PyObject *kwd
     return ret;
 }
 
+static PyObject *
+TreeSequence_ld_matrix(TreeSequence *self, PyObject *args, PyObject *kwds,
+    two_locus_count_stat_method *method)
+{
+    PyObject *ret = NULL;
+    static char *kwlist[]
+        = { "sample_set_sizes", "sample_sets", "row_sites", "col_sites", "mode", NULL };
+
+    PyObject *row_sites = NULL;
+    PyObject *col_sites = NULL;
+    PyObject *sample_set_sizes = NULL;
+    PyObject *sample_sets = NULL;
+    PyArrayObject *sample_set_sizes_array = NULL;
+    PyArrayObject *sample_sets_array = NULL;
+    PyArrayObject *row_sites_array = NULL;
+    PyArrayObject *col_sites_array = NULL;
+    PyArrayObject *result_matrix = NULL;
+    npy_intp result_shape[3];
+    char *mode = NULL;
+    tsk_size_t num_sample_sets;
+    tsk_flags_t options = 0;
+    int err;
+
+    if (TreeSequence_check_state(self) != 0) {
+        goto out;
+    }
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OOOOs", kwlist, &sample_set_sizes,
+            &sample_sets, &row_sites, &col_sites, &mode)) {
+        goto out;
+    }
+    if (parse_stats_mode(mode, &options) != 0) {
+        goto out;
+    }
+    if (parse_sample_sets(sample_set_sizes, &sample_set_sizes_array, sample_sets,
+            &sample_sets_array, &num_sample_sets)
+        != 0) {
+        goto out;
+    }
+    row_sites_array = (PyArrayObject *) PyArray_FROMANY(
+        row_sites, NPY_INT32, 1, 1, NPY_ARRAY_IN_ARRAY);
+    if (row_sites_array == NULL) {
+        goto out;
+    }
+    col_sites_array = (PyArrayObject *) PyArray_FROMANY(
+        col_sites, NPY_INT32, 1, 1, NPY_ARRAY_IN_ARRAY);
+    if (col_sites_array == NULL) {
```

Comment:
Not testing this case - need to pass in some bad arguments to ``col_sites``, like ``col_sites = ["asdf"]`` or something.